src/__tests__/utils/url_utils.test.js

```diff
@@ -0,0 +1,58 @@
+import { getOriginFromUrl, getLocationFromUrl } from '../../utils/url_utils';
+
+describe('url utils', () => {
+  describe('getOriginFromUrl', function() {
+    it('should return undefined if there is no url', function() {
+      expect(getOriginFromUrl()).toBe(undefined);
+      expect(getOriginFromUrl('')).toBe(undefined);
+      expect(getOriginFromUrl(null)).toBe(undefined);
+    });
+    it('should use an anchor to parse the url and return the origin', function() {
+      var url = 'https://test.com/example';
+      expect(getOriginFromUrl(url)).toBe('https://test.com');
+    });
+    it('should use add the `port` when available', function() {
+      var url = 'https://localhost:3000/example';
+      expect(getOriginFromUrl(url)).toBe('https://localhost:3000');
+    });
+  });
+  describe('getLocationFromUrl', function() {
+    const mapping = {
+      'https://localhost:3000/foo?id=1': {
+        href: 'https://localhost:3000/foo?id=1',
+        protocol: 'https:',
```

Comment:
protocol should not include the `:`.

Reply:
Actually, the function does return the protocol with a colon:

```
// http://banana.com
host:"banana.com"
hostname:"banana.com"
href:"http://banana.com"
protocol:"http:"
```

[Wikipedia says](https://en.wikipedia.org/wiki/URL#Syntax) that protocol is:

> A non-empty scheme component followed by a colon

... which tells me that protocol should not include it. That said, type this into your console:

```
> window.location
...
< protocol: "https:"
```

The built-in `URL` object (MDN) has the colon so I think this is the right implementation. Again, in the console:

```
> var url = new URL(window.location.toString()); url;
...
< protocol: "https:"
```

Reply:
lucho is discovering all the greatness or javascript. our boy is growing :tada